engine/Default/cargo_dump_processing.php

```diff
@@ -31,39 +31,53 @@
 	create_error('You can\'t dump cargo in a Federal Sector!');
 }
 
-require_once('shop_goods.inc');
+$container = create_container('skeleton.php', 'current_sector.php');
+
+if ($player->getExperience() > 0) {
+	// If they have any experience left, lose exp
+
+	// get the distance
+	$x = Globals::getGood($good_id);
+	$x['TransactionType'] = 'Sell';
+	$good_distance = Plotter::findDistanceToX($x, $sector, true);
+	if(is_object($good_distance)) {
+		$good_distance = $good_distance->getRelativeDistance();
+	}
+	$good_distance = max(1,$good_distance);
+
+	// Don't lose more exp than you have
+	$lost_xp = min($player->getExperience(),
+	               round(SmrPort::getBaseExperience($amount, $good_distance)));
+	$player->decreaseExperience($lost_xp);
+	$player->increaseHOF($lost_xp, array('Trade','Experience', 'Jettisoned'), HOF_PUBLIC);
 
-// get the distance
-$x = Globals::getGood($good_id);
-$x['TransactionType'] = 'Sell';
-$good_distance = Plotter::findDistanceToX($x, $sector, true);
-if(is_object($good_distance)) {
-	$good_distance = $good_distance->getRelativeDistance();
+	$container['msg'] = 'You have jettisoned <span class="yellow">'.$amount.'</span> ' . pluralise('unit', $amount) . ' of '.$good_name.' and have lost <span class="exp">'.$lost_xp.'</span> experience.';
+	// log action
+	$account->log(LOG_TYPE_TRADING, 'Dumps '.$amount.' of '.$good_name.' and loses '.$lost_xp.' experience', $player->getSectorID());
 }
-$good_distance = max(1,$good_distance);
+else {
+	// No experience to lose, so damage the ship
+	$damage = round($amount/5);
```

Comment:
This should maybe be `ceiling` or have a special message when you take 0 damage (eg 1-2 goods jettisoned) - I'm not fussed though just noticed it

Reply:
Okay, I changed the `round` to a `ceil`.